elastic/kibana
# [Ingest Pipelines] Add license checks for processors

## Summary

This PR adds support for allowing certain processors to be available depending on license type. And more specifically it already makes the `redact` and `inference` processors be available only if license >= `platinum`.

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Ingest Pipelines` and create a new pipeline
3. Open up the dropdown for selecting processors and make sure that the `redact` and `inference` processors are shown
4. Go to the `License Management` app and revert the license to `Basic`
5. Navigate to `Stack Management` -> `Ingest Pipelines` and create a new pipeline
6. Check that the dropdown for selecting processors doesnt show `redact` and `inference` processors

Closes https://github.com/elastic/kibana/issues/153557
